### ISSUE TITLE
Rewrite routes with path outside project root 

### ISSUE THREAD

**Issue by binaryben** [vuejs/vitepress#2335]
### Describe the bug

Not sure if a bug or user error. Should this work? (check reproduction)

### Reproduction

```ts
const githubRepo = resolve(__dirname, '..', '..', '.github', 'docs')
let rewrites = {}
rewrites[resolve(githubRepo, 'SECURITY.md')] = 'policy/security'

export default defineConfig({
  // ...
  rewrites,
  // ...
})
```

### Expected behavior

Navigating to https://siteurl/policy/security should load content from the .github repo source

### System Info

```sh
System:
    OS: macOS 13.3.1
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Memory: 68.12 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.9.1 - ~/.nvm/versions/node/v18.9.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.9.1/bin/yarn
    npm: 9.4.0 - ~/.nvm/versions/node/v18.9.1/bin/npm
  Browsers:
    Chrome: 112.0.5615.137
    Safari: 16.4
  npmPackages:
    vitepress: 1.0.0-alpha.74 => 1.0.0-alpha.74
```

### Additional context

A quick `console.log` of the config shows the correct path is loaded under the rewrite object.

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by binaryben** [comment]
Small update after further debugging effort...

I tried a more minimal example. Added `test.md` to my `srcDir` root and added the following:

```
rewrites['test.md'] = 'foobar.md'
```

This works after I noticed the docs have file extensions on the from and destination path 😅. I tried adding the .md extension to the destination path to see if it would fix it:

```
rewrites[resolve(githubRepo, 'SECURITY.md')] = 'policy/security.md'
```

Unfortunately no luck. Fair to assume the rewrite is happening in the Vue router layer with no context of file paths outside the browser?
